homeassistant/components/ws66i/__init__.py

```diff
@@ -0,0 +1,62 @@
+"""The Soundavo WS66i 6-Zone Amplifier integration."""
+import logging
+
+from pyws66i import get_ws66i
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_IP_ADDRESS
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .config_flow import validate_input
+from .const import CONF_NOT_FIRST_RUN, DOMAIN, FIRST_RUN, WS66I_OBJECT
+
+PLATFORMS = ["media_player"]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Soundavo WS66i 6-Zone Amplifier from a config entry."""
+
+    # double negative to handle absence of value
+    first_run = not bool(entry.data.get(CONF_NOT_FIRST_RUN))
+    if first_run:
+        hass.config_entries.async_update_entry(
+            entry, data={**entry.data, CONF_NOT_FIRST_RUN: True}
+        )
```

Comment:
Do the first run when the device is setup in the config entry so we don't have to do it here and then we don't have to keep track of this

Reply:
I got rid of it. It was used to know whether to update entities before registration in order to know which entities to enable/disable. I made some improvements there where the "first run" is not needed.